battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java

```diff
@@ -52,4 +56,49 @@ public void testIfPlayerTermAreCorrect(){
 
         assertTrue(status.status.equals(MovementStatus.SUCCESS));
     }
+    
+    @DataProvider
+    public Object[][] shootingList (){
+        return new Object[][] {
+                {new LinkedList<Integer>(Arrays.asList(1, 2, 3, 4, 5, 6, 7))},
+                {new LinkedList<Integer>(Arrays.asList(10, 2, 35, 42, 51, 68, 77))},
+                {new LinkedList<Integer>(Arrays.asList(11, 22, 33, 44, 55, 66, 73))},
+                {new LinkedList<Integer>(Arrays.asList(15, 12, 53, 84, 51, 62, 99))}
+        };
+    }
+    @Test(dataProvider = "shootingList")
+    public void testPlayerSalvaShooting(List<Integer> shootingList) {
+        Playing playing = new Playing();
+        Registration registration = mock(Registration.class);
+        when(registration.getPlayer()).thenReturn(mock(Player.class));
+
+        PlacingShip placingShip = mock(PlacingShip.class);
+        when(placingShip.myManager()).thenReturn(mock(DamageManager.class));
+
+        PlayerOrganizer organizer = mock(PlayerOrganizer.class);
+        GameController controller = mock(GameController.class);
+
+        for(int shoot: shootingList) {
+            when(controller.salvaShoot(eq(shoot), any(Player.class))).thenReturn(MovementStatus.TRY_AGAIN);
+        }
+
+        when(controller.getWinner()).thenReturn(null);
+        when(organizer.myController()).thenReturn(controller);
+
+        MockHttpSession session = new MockHttpSession();
+        session.setAttribute("registration", registration);
+        session.setAttribute("placingShip", placingShip);
+        session.setAttribute("playerOrganizer", organizer);
+        playing.session = session;
+
+        try {
+            playing.afterPropertiesSet();
```

Comment:
Think about adding 'throws Exception' to method signature and let testNG to handle Exception.